packages/transitive-overlay/src/util.ts

```diff
@@ -356,73 +360,79 @@ export function itineraryToTransitive(
       };
 
       // Add the "from" end of transit legs to the list of stops.
-      const fromStop = makeStop(leg.from, hasLegGeometry && legCoords[0]);
+      const fromStop = makeStop(
+        mutableLeg.from,
+        hasLegGeometry && legCoords[0]
+      );
       addStop(fromStop, newStops, knownStopNames);
-      pattern.stops.push({ stop_id: getStopId(leg.from) });
+      pattern.stops.push({ stop_id: getStopId(mutableLeg.from) });
 
       // add intermediate stops to stops dictionary and pattern object
       // If there is no intermediateStopGeometry, do not add the intermediate stops
       // as it will be straight lines instead of the nice legGeometry (but only if
       // the legGeometry exists).
       if (
-        leg.intermediateStops &&
+        mutableLeg.intermediateStops &&
         (hasIntermediateStopGeometry || !hasLegGeometry)
       ) {
-        leg.intermediateStops.forEach((stop, i) => {
+        mutableLeg.intermediateStops.forEach((stop, i) => {
           // FIXME: line up the coordinates of the stops so they appear on the line.
           addStop(stop, newStops, knownStopNames);
           pattern.stops.push({
             // Intermediate stops are still exclusively using stopId.
             stop_id: stop.stopId,
             geometry:
-              hasIntermediateStopGeometry && leg.interStopGeometry[i].points
+              hasIntermediateStopGeometry &&
+              mutableLeg.interStopGeometry[i].points
           });
         });
       }
 
       // Add the "to" end of transit legs to the list of stops.
       // (Do not label stop names if they repeat.)
       const lastCoord = hasLegGeometry && legCoords[legCoords.length - 1];
-      const toStop = makeStop(leg.to, lastCoord);
+      const toStop = makeStop(mutableLeg.to, lastCoord);
```

Comment:
I would create a method that clones the original `to` stop and includes the flex stuff from lines 328/330. The main benefit is that you don't rename/clone the entire leg just to change a small piece of it, given that the cloned/modified leg is not used anywhere else other than here.